src/app_service/service/message/service.nim

```diff
@@ -793,13 +793,16 @@ QtObject:
       var chatId: string
       discard responseObj.getProp("chatId", chatId)
 
-      var messageId = ""
+      var messageId: string
 
       if(error.len > 0):
         error "error: ", procName="onGetFirstUnseenMessageIdFor", errDescription=error
       else:
         discard responseObj.getProp("messageId", messageId)
 
+      if(messageId.len == 0):
```

Comment:
This seems to be the root of the problem in the current state, with self.initialMessagesLoaded state removed. I believe this is what I should pursue next.

Reply:
We should probably still send an event. Either we send a `SIGNAL_FIRST_UNSEEN_MESSAGE_FAILED` event, or we just use the same event as below and just have the `messageId.len == 0` condition in the module.